web/impact/impact/tests/test_algolia_api_key_view.py

```diff
@@ -128,7 +174,7 @@ def test_alumni_user_only_sees_mentors_of_alumni_programs(self):
 
                 for program in programs:
                     self.assertNotIn(program.name, response_data["filters"])
-                self.assertIn(other_program.name, response_data["filters"])
+                self.assertIn(alumni_program.name, response_data["filters"])
```

Comment:
The name of this test is `test_user_with_staff_role_grant_sees_all_mentors`. I'm not sure I see how this relates to that name. It's a very confusing test and I'm not sure in fact what it's meant to be testing at all. It seems to create a lot of supporting objects, but very little of that is used in the asserts. 
Can it be cleaned up so the next person has a chance of understanding what's going on here?

Reply:
The test is `test_alumni_user_only_sees_mentors_of_alumni_programs`, and we create a bunch of ended programs, give the user role grants to one of them, as well as to an active alumni program. Making sure they only see alumni things.

Reply:
okay, still seems kind of messy but I'll let it go